streams/src/main/java/org/apache/kafka/streams/processor/internals/StateUpdater.java

```diff
@@ -170,6 +174,27 @@ public int hashCode() {
      */
     Set<Task> getTasks();
 
+    /**
+     * Gets all tasks that are currently being restored inside the state updater.
+     *
+     * Tasks that have just being added into the state updater via {@link StateUpdater#add(Task)}
+     * or have restored completely or removed will not be returned; similarly tasks that have just being
+     * removed via {@link StateUpdater#remove(TaskId)} maybe returned still.
+     *
+     * @return set of all updating tasks inside the state updater
+     */
+    Set<Task> getUpdatingTasks();
+
+    /**
+     * Gets all tasks that are paused from restoring inside the state updater.
+     *
+     * Tasks that have just being paused in the state updater via {@link StateUpdater#pause(TaskId)}
+     * or have restored completely or removed will not be returned.
+     *
+     * @return set of all tasks paused inside the state updater
+     */
+    Set<Task> getPausedTasks();
+
```

Comment:
We do not need to expose those anymore, right?

Reply:
I thought about this a bit, and I think it's still valuable for 1) testing purposes, and 2) metrics in the future.

Reply:
I do not think we those for testing purposes since you usually test the implementation of an interface and not the interface itself. For exmple, on `DefaultStateUpdater` that is an implementation of this interface those APIs are exposed for testing purposes. I am also not convinced about metrics since those should be recorded inside the state updater implementation. I propose to remove those APIs and add them later if needed.

Reply:
Cool, I think I'm convinced, I will remove those APIs then.